REST/Listener.cc

```diff
@@ -98,6 +104,40 @@ namespace litecore { namespace REST {
         return false;
     }
 
+    bool Listener::registerCollection(const string& name, C4CollectionSpec collection) {
+        lock_guard<mutex> lock(_mutex);
+        auto i = _databases.find(name);
+        if (i == _databases.end())
+            return false;
+
+        auto j = _allowedCollections.find(name);
+        if(j == _allowedCollections.end()) {
+            vector<C4CollectionSpec> collections({collection});
+            _allowedCollections.emplace(name, collections);
+        } else {
+            j->second.push_back(collection);
+        }
+        
+        return true;
+    }
+
+
+    bool Listener::unregisterCollection(const string& name, C4CollectionSpec collection) {
```

Comment:
Use CollectionSpec for consistent with other C++ code? It also implements == operator.